File: OPENAPI-REST-API/openapi-client/javascript-closure-angular/API/Client/MailServiceType.js
goog.provide('API.Client.MailServiceType');

/**
 * Service type definition for a mail service, including name, pricing, and category.
 * @record
 */
API.Client.MailServiceType = function() {}

/**
 * The ID of the service type.
 * @type {!string}
 * @export
 */
API.Client.MailServiceType.prototype.servicesId;

/**
 * The name of the service type.
 * @type {!string}
 * @export
 */
API.Client.MailServiceType.prototype.servicesName;

/**
 * The cost of the service type.
 * @type {!string}
 * @export
 */
API.Client.MailServiceType.prototype.servicesCost;

/**
 * The category of the service type.
 * @type {!string}
 * @export
 */
API.Client.MailServiceType.prototype.servicesCategory;

/**
 * Whether the service type is buyable.
 * @type {!string}
 * @export
 */
API.Client.MailServiceType.prototype.servicesBuyable;

/**
 * The type of the service type.
 * @type {!string}
 * @export
 */
API.Client.MailServiceType.prototype.servicesType;

/**
 * Additional field for the service type.
 * @type {!string}
 * @export
 */
API.Client.MailServiceType.prototype.servicesField1;

/**
 * Additional field for the service type.
 * @type {!string}
 * @export
 */
API.Client.MailServiceType.prototype.servicesField2;

/**
 * The module of the service type.
 * @type {!string}
 * @export
 */
API.Client.MailServiceType.prototype.servicesModule;

File: OPENAPI-REST-API/swagger-client/csharp-dotnet2/src/main/CsharpDotNet2/myadmin-client-csharp-dotnet2/Model/MailServiceType.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace IO.Swagger.Model {

  /// <summary>
  /// Service type definition for a mail service, including name, pricing, and category.
  /// </summary>
  [DataContract]
  public class MailServiceType {
    /// <summary>
    /// The ID of the service type.
    /// </summary>
    /// <value>The ID of the service type.</value>
    [DataMember(Name="services_id", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "services_id")]
    public string ServicesId { get; set; }

    /// <summary>
    /// The name of the service type.
    /// </summary>
    /// <value>The name of the service type.</value>
    [DataMember(Name="services_name", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "services_name")]
    public string ServicesName { get; set; }

    /// <summary>
    /// The cost of the service type.
    /// </summary>
    /// <value>The cost of the service type.</value>
    [DataMember(Name="services_cost", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "services_cost")]
    public string ServicesCost { get; set; }

    /// <summary>
    /// The category of the service type.
    /// </summary>
    /// <value>The category of the service type.</value>
    [DataMember(Name="services_category", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "services_category")]
    public string ServicesCategory { get; set; }

    /// <summary>
    /// Whether the service type is buyable.
    /// </summary>
    /// <value>Whether the service type is buyable.</value>
    [DataMember(Name="services_buyable", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "services_buyable")]
    public string ServicesBuyable { get; set; }

    /// <summary>
    /// The type of the service type.
    /// </summary>
    /// <value>The type of the service type.</value>
    [DataMember(Name="services_type", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "services_type")]
    public string ServicesType { get; set; }

    /// <summary>
    /// Additional field for the service type.
    /// </summary>
    /// <value>Additional field for the service type.</value>
    [DataMember(Name="services_field1", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "services_field1")]
    public string ServicesField1 { get; set; }

    /// <summary>
    /// Additional field for the service type.
    /// </summary>
    /// <value>Additional field for the service type.</value>
    [DataMember(Name="services_field2", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "services_field2")]
    public string ServicesField2 { get; set; }

    /// <summary>
    /// The module of the service type.
    /// </summary>
    /// <value>The module of the service type.</value>
    [DataMember(Name="services_module", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "services_module")]
    public string ServicesModule { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class MailServiceType {\n");
      sb.Append("  ServicesId: ").Append(ServicesId).Append("\n");
      sb.Append("  ServicesName: ").Append(ServicesName).Append("\n");
      sb.Append("  ServicesCost: ").Append(ServicesCost).Append("\n");
      sb.Append("  ServicesCategory: ").Append(ServicesCategory).Append("\n");
      sb.Append("  ServicesBuyable: ").Append(ServicesBuyable).Append("\n");
      sb.Append("  ServicesType: ").Append(ServicesType).Append("\n");
      sb.Append("  ServicesField1: ").Append(ServicesField1).Append("\n");
      sb.Append("  ServicesField2: ").Append(ServicesField2).Append("\n");
      sb.Append("  ServicesModule: ").Append(ServicesModule).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}
